src/extension/alterschema/linter/non_applicable_enum_validation_keywords.h

```diff
@@ -0,0 +1,83 @@
+class NonApplicableEnumValidationKeywords final : public SchemaTransformRule {
+public:
+  NonApplicableEnumValidationKeywords()
+      : SchemaTransformRule{
+            "enum_validation_keywords_default",
+            "Setting validation keywords that do not apply to any item in "
+            "`enum` is considered an "
+            "anti-pattern, as the enumeration choices already imply their "
+            "respective types"} {};
+
+  [[nodiscard]] auto
+  condition(const sourcemeta::core::JSON &schema,
+            const sourcemeta::core::JSON &,
+            const sourcemeta::core::Vocabularies &vocabularies,
+            const sourcemeta::core::SchemaFrame &,
+            const sourcemeta::core::SchemaFrame::Location &,
+            const sourcemeta::core::SchemaWalker &walker,
+            const sourcemeta::core::SchemaResolver &) const
+      -> sourcemeta::core::SchemaTransformRule::Result override {
+    if (!contains_any(vocabularies,
+                      {"https://json-schema.org/draft/2020-12/vocab/validation",
+                       "https://json-schema.org/draft/2019-09/vocab/validation",
+                       "http://json-schema.org/draft-07/schema#",
+                       "http://json-schema.org/draft-06/schema#",
+                       "http://json-schema.org/draft-04/schema#",
+                       "http://json-schema.org/draft-03/schema#",
+                       "http://json-schema.org/draft-02/schema#",
+                       "http://json-schema.org/draft-02/hyper-schema#",
+                       "http://json-schema.org/draft-01/schema#",
+                       "http://json-schema.org/draft-01/hyper-schema#"}) ||
+        !schema.is_object() || !schema.defines("enum") ||
+        !schema.at("enum").is_array()) {
+      return false;
+    }
+
+    std::set<sourcemeta::core::JSON::Type> enum_types;
+    for (const auto &value : schema.at("enum").as_array()) {
+      enum_types.emplace(value.type());
+    }
+
+    if (enum_types.empty()) {
+      return false;
+    }
+
+    this->blacklist.clear();
+    for (const auto &entry : schema.as_object()) {
+      const auto metadata = walker(entry.first, vocabularies);
+      if (metadata.type == sourcemeta::core::SchemaKeywordType::Other ||
+          metadata.type == sourcemeta::core::SchemaKeywordType::Reference) {
+        continue;
+      }
+
+      if (metadata.instances.empty()) {
+        continue;
+      }
+
+      if (std::ranges::none_of(metadata.instances.cbegin(),
+                               metadata.instances.cend(),
+                               [&enum_types](const auto keyword_type) {
+                                 return enum_types.contains(keyword_type);
+                               })) {
+        this->blacklist.emplace_back(entry.first);
+      }
+    }
+
+    if (this->blacklist.empty()) {
+      return false;
+    }
+    std::ostringstream message;
```

Comment:
```suggestion

    std::ostringstream message;
```

Just for consistency